visualization-design.typ

```diff
@@ -105,15 +105,137 @@ room = {
+
+Next, add the job to your `prometheus.yml` file. Example:
+
+```yaml
+- job_name: 'grafana_metrics'
+
+   scrape_interval: 15s
+   scrape_timeout: 5s
+
+   static_configs:
+     - targets: ['localhost:3000']
+```
+
+This example job instructs Prometheus to scrape from port 3000 every 15 seconds. After adding the job, restart Prometheus and it should appear on the targets tab.
+
+Finally in Grafana, hover your mouse over the Configuration (gear) icon on the left sidebar and then click Data Sources. Select the Prometheus data source, and on the Dashboards tab, import the Grafana metrics dashboard. All scaped metrics should now be available.
+
+=== Setup Instructions From Prometheus
+
+To create a Prometheus data source in Grafana:
+
+- Click on the "cogwheel" in the sidebar to open the Configuration menu.
+- Click on "Data Sources".
+- Click on "Add data source".
+- Select "Prometheus" as the type.
+- Set the appropriate Prometheus server URL (for example, http://localhost:9090/)
+- Adjust other data source settings as desired (for example, choosing the right Access method).
+- Click "Save & Test" to save the new data source.
 
-- Querying Prometheus with Grafana: Grafana supports querying Prometheus, and there is documentation linked below on a quick start for creating a new data source. Given the balancer has already integrated Prometheus for metrics, this is the preferred method for gathering data. There is no mention on compatibility with D3.js, but assuming there are no issues integrating D3.js into a Grafana panel, this should not be a problem. #cite(<grafana-prometheus-visualization>)
+To create a new Grafana graph:
+
+- Click the graph title, then click "Edit".
+- Under the "Metrics" tab, select your Prometheus data source (bottom right).
+- Enter any Prometheus expression into the "Query" field, while using the "Metric" field to lookup metrics via autocompletion.
+- To format the legend names of time series, use the "Legend format" input. For example, to show only the method and status labels of a returned query result separated by a dash, you could use the legend format string {{method}} - {{status}}.
+- Tune other graph settings until you have a working graph.
+
+== Recieving Information From Load Balancers
+
+Grafana is a tool primarily meant for time series data, and no current data source plugins support recieving the type of information needed. To add support a custom data source must be created
+
+First, create a new plugin. There are a few different types of data source plugins that can be developed for Grafana, but a backend plugin has the most additional capabilities. A tutorial for setting up plugins can be found here: #link("https://grafana.com/developers/plugin-tools/tutorials/build-a-data-source-backend-plugin")
+
+Once the plugin is working, a data query will have to be written in Go:
```

Comment:
What if we just created a new rust crate instead? We could just serve the collected system state info over http and fetch it using the js part of the plugin. We'd be able to reuse the monolith discovery code for discovering balancers, and we wouldn't have to update all the repo infra to support go.

Reply:
I didn't know that was an option. If you think it'd be a better way to fetch data I'm all for it

Reply:
The biggest problem I foresee is that it would make the deployment for visualizer a bit more complicated, but I think that's a small compromise compared to getting everyone to learn a new language.